internal/tasks/fleet_selector.go

```diff
@@ -450,19 +456,30 @@ func (f FleetSelectorMatchingLogic) findDeviceOwnerAmongAllFleets(ctx context.Co
 			matchingFleets = append(matchingFleets, *fleet.Metadata.Name)
 		}
 	}
-	newAnnotationValue := CreateOverlappingAnnotationValue(matchingFleets)
-	currentAnnotationValue := GetOverlappingAnnotationValue(device.Metadata.Annotations)
+	newConditionMessage := createOverlappingConditionMessage(matchingFleets)
+	currentConditionMessage := ""
+	condition := api.FindStatusCondition(device.Status.Conditions, api.DeviceMultipleOwners)
+	if condition != nil {
+		currentConditionMessage = condition.Message
+	}
 
 	err := f.setDeviceOwnerAccordingToMatchingFleets(ctx, device, currentOwnerFleet, matchingFleets)
 	if err != nil {
 		return nil, err
 	}
 
-	if currentAnnotationValue != newAnnotationValue {
-		annotations := map[string]string{
-			model.DeviceAnnotationMultipleOwners: newAnnotationValue,
+	f.log.Infof("HERE %s: %s - %s", *device.Metadata.Name, currentConditionMessage, newConditionMessage)
+	if currentConditionMessage != newConditionMessage {
+		condition := api.Condition{Type: api.DeviceMultipleOwners}
+		if len(matchingFleets) > 1 {
+			condition.Status = api.ConditionStatusTrue
+			condition.Reason = "MultipleOwners"
```

Comment:
should the type be a higher level condition just feels extremely specific. For example. 

DeviceVerification
False
MultipleOwners
Msg

Reply:
I see your point, but I'd like to make it clearer that the condition relates to the resources in the DB and not the actual device.  Also, "Device" in "DeviceVerification" feels redundant.

The device already has a "ValidSpec" condition.  If I generalize both of them, we want to know if the service is offering the latest spec to the agent.  If a device has multiple owners, the fleet logic won't even update the device's spec.  If the device.spec is invalid, the device logic won't offer the spec to the agent.  I can't think of a good name for this, but maybe that should be the scope of the condition.

Reply:
@hexfusion WDYT?

Reply:
I am fine with it as is just something to think about.